src/hb-ot-var-common.hh

```diff
@@ -914,6 +914,99 @@ struct tuple_delta_t
     }
     return encoded_len;
   }
+
+  bool calc_inferred_deltas (const contour_point_vector_t& orig_points)
+  {
+    unsigned point_count = orig_points.length;
+    if (point_count != indices.length)
+      return false;
+
+    unsigned ref_count = 0;
+    hb_vector_t<unsigned> end_points;
+
+    for (unsigned i = 0; i < point_count; i++)
+    {
+      if (indices.arrayZ[i])
+        ref_count++;
+      if (orig_points.arrayZ[i].is_end_point)
+        end_points.push (i);
+    }
+    /* all points are referened, nothing to do */
+    if (ref_count == point_count)
+      return true;
+    if (unlikely (end_points.in_error ())) return false;
+
+    hb_set_t inferred_idxes;
```

Comment:
Any chance the IUP code can be shared with gvar table?

Reply:
yes, once initial implementation is done, I will try to work on sharing iup code as much as possible.

Reply:
> yes, once initial implementation is done, I will try to work on sharing iup code as much as possible.

Sounds good.